### ISSUE TITLE
OV5642 - Invalid JPEG image data

### ISSUE THREAD

**Issue by CircuitDesigners** [ArduCAM/Arduino#393]
Hi,

I am using OV5642 Camera module REV B (http://www.arducam.com/products/camera-breakout-board/5mp-ov5642/).

I am receiving image data begging from FFD8 and ending with FFD9 and when i print it through serial monitor i came to know that its a invalid data. So i create a .jpeg image file with that hex data printed on UART using HxD software. Then i open the image in windows10 and i found the image is Totally grey.

I am using STM32F429ZI discovery board.

I am attaching image and its hex data file.

-> IMAGE FILE (.jpg)
![1](https://user-images.githubusercontent.com/43670996/53889942-4bce9500-404d-11e9-9a84-2274b6f35068.jpg)

->IMAGE HEX DATA FILE
[1.txt](https://github.com/ArduCAM/Arduino/files/2936761/1.txt)

Please help me to solve this issue. And also let me know if you need further information from my side.

Thanks.

**Comment by UCTRONICS** [comment]
@CircuitDesigners 
I guess the camera doesn't init normally. Please refer to our camera configuration on STM32F03 platform
https://github.com/ArduCAM/STM32/blob/master/HardWare/ArduCAM.c

**Comment by CircuitDesigners** [comment]
@UCTRONICS 

I exactly do what has been done in library you suggested above and i am getting all data ZERO (i mean not getting anything).

I did in following manner:
I convert all below function with your register settings mentioned in "ov5642_regs.h"
1) wrSensorReg16_8(0x3008, 0x80);
2) wrSensorRegs16_8(OV5642_QVGA_Preview);
3) wrSensorRegs16_8(OV5642_JPEG_Capture_QSXGA);
4) wrSensorRegs16_8(ov5642_320x240);
5) wrSensorReg16_8(0x3818, 0xa8);
6) wrSensorReg16_8(0x3621, 0x10);
7) wrSensorReg16_8(0x3801, 0xb0);
8) wrSensorReg16_8(0x4407, 0x04);
9) uint8_t _x3503;
10) wrSensorReg16_8(0x5001,_x3503|0x01);	 // Close auto exposure mode
11) wrSensorReg16_8(0x3500,0x00);
12) wrSensorReg16_8(0x3501,0x79);
13) wrSensorReg16_8(0x3502,0xe0);	

What could be the problem ?

Thanks.

**Comment by UCTRONICS** [comment]
@CircuitDesigners 
Firstly, you should ensure the i2c bus works normally.
Notice: After you have written the value to the sensor register, it is necessary to add some delay.
If not, the sensor will not init normally.

**Comment by CircuitDesigners** [comment]
@UCTRONICS 
I added delay also between register settings. And i am sure that i2c is working properly as i am able to read chip id and also if i apply my old settings in same code then ill get result as mentioned above (1st message).

**Comment by UCTRONICS** [comment]
@CircuitDesigners 
Can you attach me an i2c wave when you write and read sensor register?

**Comment by CircuitDesigners** [comment]
@UCTRONICS 
Sorry as i am not having oscilloscope. But if you want then i can attach my i2c file and result of my serial monitor.

**Comment by CircuitDesigners** [comment]
@UCTRONICS 
Help me to solve this. Please.

**Comment by UCTRONICS** [comment]
@CircuitDesigners 
Please send me your whole project and I will help you check it.

**Comment by CircuitDesigners** [comment]
@UCTRONICS 
i am mentioning link for my whole code. Please look into it and let me know.

https://drive.google.com/open?id=1Ke7WtVtc8AcJCJEULgtvzsIJPlqXw2Eb

Thanks

**Comment by CircuitDesigners** [comment]
@UCTRONICS 
Hi, Did you find my code ? Any solution that we can discuss ?

Thanks

**Comment by UCTRONICS** [comment]
@CircuitDesigners 
Yes, I have downloaded your code, and our engineer is checking your code. 
Please be patient. We've been busy lately.

**Comment by UCTRONICS** [comment]
@CircuitDesigners 
I advise you to test it on Arduino UNO platform firstly. You need to ensure the camera module works normally.

**Comment by CircuitDesigners** [comment]
@UCTRONICS 
Unfortunately i dont have Arducam mini OV5642 neither i have Arducam shield. So its likely not possible to test OV5642 with Arduino. Please if you can, look into my code and settings and all and provide me some solution. I am waiting for you reply.

Thanks

**Comment by UCTRONICS** [comment]
@CircuitDesigners 
Please read my code in detail. https://github.com/ArduCAM/STM32

![test](https://user-images.githubusercontent.com/27711713/54574516-9677e600-4a2b-11e9-873f-3bfc4a0978cf.jpg)

**Comment by CircuitDesigners** [comment]
@UCTRONICS 
I also tried without putting markers (i remove wherever it was), but same result.

**Comment by CircuitDesigners** [comment]
@UCTRONICS 
Any update on solution ?

**Comment by Kyanaam** [comment]
I am also facing the same problem. Do we have any solution?

**Comment by UCTRONICS** [comment]
@Kyanaam 
Please attach your image here.

**Comment by Kyanaam** [comment]
![20160125023557](https://user-images.githubusercontent.com/49593370/56860588-d67ca200-69b5-11e9-8b00-066f0fe8f2ca.jpg)
I am using BeagleBone Black board. SPI/I2C working fine.

**Comment by UCTRONICS** [comment]
@Kyanaam 
I don't have this board. Have you tested the camera on Arduino platform?

**Comment by Kyanaam** [comment]
No I haven't tested this on Arduino.

**Comment by UCTRONICS** [comment]
@Kyanaam 
For SPI camera, We have a project on stm32 platform. You can get it here https://github.com/ArduCAM/RaspberryPi/tree/master/SPI_Camera

**Comment by Kyanaam** [comment]
When I use this initialization, I don't get any valid Jpeg image.
